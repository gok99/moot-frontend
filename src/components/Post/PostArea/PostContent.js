import React, { useState } from 'react';
import { Row, Col, Button } from 'react-bootstrap';

import '../../Styles/styles.css';
import '../post.css';

const PostContent = (props) => {
  const noPost = props.noPost;
  const myPost = props.myPost;
  const postTime = props.postTime;
  const postTitle = props.postTitle;
  const postContent = props.postContent;
  const posterUid = props.posterUid;
  const friends = props.friends;
  const initialPostContentState = props.initialPostContentState;
  const [postContentState, setPostContentState] = useState(initialPostContentState);
  var posterState = "an anonymous user";
  for (let friend of friends) {
    if (friend.friendUid === posterUid) {
      posterState = friend.username;
    }
  }

  const toggleExpand = () => {
    setPostContentState(!postContentState);
  }

  return (
    <>
      <Row>
        <Col className="d-flex justify-content-start">     
          { noPost
            ? <p className="text-post meta">No more posts...<br /><br /></p>
            : myPost
              ? <p className="text-post meta">Posted by me on {postTime}<br /><br /></p>
              : <p className="text-post meta">Posted by {posterState} on {postTime}<br /><br /></p>
          }
        </Col>
      </Row>
      <Row>
        <p className="text-post title">{ postTitle }</p>
      </Row>
      <br />
      <Row className={postContentState ? "b-text-postcontent-expand" : "b-text-postcontent"}>
        <p className="text-post content">{ postContent }</p>
      </Row>
      <Row className="d-flex justify-content-end">
        <Button className="btn-postexpand mt-3" type="button" onClick={toggleExpand}>
          {postContentState ? "Show Less..." : "Expand..."}
        </Button>
      </Row>
    </>
  );
};

export default PostContent;