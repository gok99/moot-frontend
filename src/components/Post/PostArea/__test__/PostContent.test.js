import React from 'react';
import ReactDOM from 'react-dom';
import PostContent from '../PostContent';

import { cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect"
import renderer from 'react-test-renderer';

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement('div');
  ReactDOM.render(<PostContent noPost={false} myPost={false} postTime={0} postTitle={"test"} postContent={"test"} posterUid={0} friends={[]}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", () => {
  const tree = renderer.create(<PostContent noPost={false} myPost={false} postTime={0} postTitle={"test"} postContent={"test"} posterUid={0} friends={[]}/>).toJSON();
  expect(tree).toMatchSnapshot();
});