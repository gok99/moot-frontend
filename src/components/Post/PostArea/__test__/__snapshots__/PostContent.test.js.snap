// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    className="row"
  >
    <div
      className="d-flex justify-content-start col"
    >
      <p
        className="text-post meta"
      >
        Posted by 
        an anonymous user
         on 
        0
        <br />
        <br />
      </p>
    </div>
  </div>,
  <div
    className="b-text-posttitle row"
  >
    <p
      className="text-post title"
    >
      test
    </p>
  </div>,
  <br />,
  <div
    className="b-text-postcontent row"
  >
    <p
      className="text-post content"
    >
      test
    </p>
  </div>,
  <div
    className="d-flex justify-content-end row"
  >
    <button
      className="btn-postexpand mt-3 btn btn-primary"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Expand...
    </button>
  </div>,
]
`;
